apps/frontend/src/lib/psi/refresh.ts

```diff
@@ -0,0 +1,242 @@
+import {
+  tensionPairs,
+  hotTakeLabels
+} from "@/common/constants";
+import { hashCommit } from "@/lib/psi/hash";
+import { BASE_API_URL } from "@/config";
+import { getConnectionSigPubKey } from "@/lib/user";
+import { socketEmit } from "@/lib/socket";
+import {
+  errorToString,
+  SocketRequestType
+} from "@types";
+import {
+  User,
+  Connection,
+  UserData,
+  Session
+} from "@/lib/storage/types";
+import { storage } from "@/lib/storage";
+import { sendMessages } from "@/lib/message";
+import { Socket } from "socket.io-client";
+import { updateUserData } from "@/lib/storage/localStorage/user/userData";
+
+export type Intersection = {
```

Comment:
This should eventually be a storage-level type once we are saving PSI results

Reply:
Agreed, as I mentioned in TG want to add that in the plaintext PSI / friend PSIs where the PSI info is actually used.